linkml/generators/prefixmapgen.py

```diff
@@ -64,11 +70,32 @@ def end_schema(self, base: Optional[str] = None, output: Optional[str] = None, *
             context[k] = v
         for k, v in self.slot_class_maps.items():
             context[k] = v
-        if output:
-            with open(output, 'w') as outf:
-                outf.write(as_json(context))
+
+        if self.output_path:
```

Comment:
The generator class doesn't need to worry about output path.

The job of any one generator is to produce the output, everything else is generic